### ISSUE TITLE
HTML filter is not applied just after allowance if strict-blocking is temporary allowed

### ISSUE THREAD

**Issue by Yuki2718** [uBlockOrigin/uBlock-issues#1507]
### Prerequisites

- [ ] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [ ] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [ ] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [ ] uBlock Origin is the only extension
    - [ ] uBlock Origin with default lists/settings
    - [ ] using a new, unmodified browser profile
- [ ] I am running the latest version of uBlock Origin
- [ ] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

It may be only me but as the title says a HTML filter is skipped just after temporal allowing of strict-blocking. Reload and the filter is correctly applied, or as long as temporal allowance is on the filter is applied - it's only just after the first allowance. If I disable the filter causing strict-blocking, again the filter is applied. And it's only on Android.

### A specific URL where the issue occurs

Google search `https://www.google.com/search?client=firefox-b-d&q=nanj-adguard&oq=nanj-adguard` -> go to `https://wikiwiki.jp/nanj-adguard/`. Tested on a few other URLs too.

### Steps to Reproduce

1. Add any valid HTML filter any rule to cause strict-blocking, for example `wikiwiki.jp##^.title2` and `||wikiwiki.jp^$doc`
2. Open the logger, visit the site, and temporary allow.
3. See HTML filter was not applied but applied if the page is reloaded.

### Expected behavior:

HTML filter is applied whatever condition is.

### Actual behavior:

Not

### Your environment

* uBlock Origin version: 1.33.2
* Browser Name and version: Firefox 85.1.3 (Build #2015792283)
* Operating System and version: Android 11

**Comment by gwarser** [comment]
Fx 85.0.2/86.0 uBO 1.33.2 on Manjaro - unable to reproduce.

**Comment by uBlock-user** [comment]
Can you repro this consistently ?

**Comment by Yuki2718** [comment]
> Can you repro this consistently ?

Yes, it's 100% times and reproduced on Firefox Beta 86.0.0-beta.5. I don't know how to take video screenshot like ghajini so at most can upload screenshots of the logger if it ever helps.

**Comment by Yuki2718** [comment]
The first SS is actually the second try - temporal allowance is still active and HTML filter is applied. I waited for the allowance to be inactive, then in the second screenshot the filter is not applied.

<details>

![Screenshot_1](https://user-images.githubusercontent.com/58900598/109184333-3b7fc980-77d2-11eb-99e7-3fe0ca88d2b8.png)

![Screenshot_2](https://user-images.githubusercontent.com/58900598/109184361-42a6d780-77d2-11eb-91ec-a8bcba225004.png)

</details>

**Comment by gwarser** [comment]
I see this with remote debugging. No errors in console.
